android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -269,18 +269,21 @@ class BrowserIcons constructor(
      * via an in-memory cache is attempted first, followed by an asynchronous load as a fallback.
      *
      * @param url The URL of the website an icon should be loaded for.
+     * @param iconResource Optional [IconRequest.Resource] to load the icon from.
      * @param iconSize The preferred size of the icon that should be loaded.
      * @param isPrivate Whether this request for this icon came from a private session.
      * @param content The Composable content block to render the icon.
      */
     @Composable
     fun LoadableImage(
         url: String,
+        iconResource: IconRequest.Resource? = null,
```

Comment:
Thanks for your suggestion @MatthewTighe! 
This is actually how I initially how I had defined the function before refactoring. The reason I chose to use the nullable `IconRequest.Resource` is because using a `List` forces the caller to know about creating a singleton list, which IMO shouldn't have an awareness of, as our caller will only ever have the single `imageUrl`.